### ISSUE TITLE
Make AutoRouteGuards work when a route is popped to

### ISSUE THREAD

**Issue by jtmuller5** [Milad-Akarie/auto_route_library#2042]
AutoRouteGuards only work for "forward" navigation. It would be awesome if we could configure each AutoRouteGuard to also work for "backward" navigation (ex. when the route on top of the protected route is popped off or when a user presses the browser back button in Flutter web).

The main use case for this is to show a confirmation popup when a user tries to exit a screen. This is useful when editing forms or other screens that can lose progress.

I created a repo that illustrates the issue:
- Press "Screen one"
- [Guard dialog appears]
- Press "Confirm"
- [Navigate to Screen Two]
- Press the back arrow
- [Guard dialog does not appear]

Sample repo (check auto_route folder): https://github.com/CodeOTR/flutter_web_navigation

In go_router, you can accomplish this by adding an [onExit](https://pub.dev/documentation/go_router/latest/go_router/GoRoute/onExit.html) function to any route.

https://github.com/user-attachments/assets/b6960657-0b98-42a4-8307-1ca631e4175b

**Comment by proninyaroslav** [comment]
Agree. I was looking for this in the documentation today, thinking that it was implemented in `AutoRouteGuard`. In my case, the authorization should expire after moving back in the stack. In the current implementation of `AutoRouteGuard`, this is not possible.
